sql/src/main/java/io/crate/planner/operators/Get.java

```diff
@@ -131,6 +132,11 @@ public ExecutionPlan build(PlannerContext plannerContext,
         );
     }
 
+    @Override
+    public long estimatedRowSize(TableStats tableStats) {
```

Comment:
I wonder if we should instead of passing the TableStats in, get the estimatedSizePerRow in the constructor and freeze it so that the LogicalPlan is a snapshot. Or pass in the internal `Stats` object.

Doing it like this the `tableStats` could be mutated, so if `estimatedRowSize` is called twice with the same `tableStats` instance one can still get different results - later on this could cause different decisions to be made which are not compatible to each other, leading to a nasty race-condition planner bug.

Reply:
Hm, originally I also didn't like the TableStats as method argument there but only because it seemed "weird" design-wise. Haven't thought of this stats issue, which adds up as an important reason to change this design.